components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -182,21 +183,25 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
             throws PostAuthenticationFailedException {
 
         String[] missingClaims = FrameworkUtils.getMissingClaims(context);
+        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK,
+                FrameworkConstants.LogConstants.ActionIDs.HANDLE_MISSING_CLAIMS)
+                .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION);
 
         if (StringUtils.isNotBlank(missingClaims[0])) {
 
             if (log.isDebugEnabled()) {
                 log.debug("Mandatory claims missing for the application : " + missingClaims[0]);
             }
             if (LoggerUtils.isDiagnosticLogsEnabled()) {
-                Map<String, Object> params = new HashMap<>();
-                params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER, context.getServiceProviderName());
-                params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, context.getTenantDomain());
-                params.put(FrameworkConstants.LogConstants.MISSING_CLAIMS, missingClaims);
-                LoggerUtils.triggerDiagnosticLogEvent(
-                        FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params, LogConstants.FAILED,
-                        "Mandatory claims missing for the application: " + context.getServiceProviderName(),
-                        FrameworkConstants.LogConstants.ActionIDs.HANDLE_MISSING_CLAIMS, null);
+                diagnosticLogBuilder.inputParam(LogConstants.InputKeys.SERVICE_PROVIDER,
```

Comment:
why we are using different keys for application? `In the previous log, it was as APPLICATION_NAME, and here it is ServiceProvider?

Reply:
This will be fixed with https://github.com/wso2/carbon-identity-framework/pull/4809